### PR TITLE
Support for TypeScript 2.6

### PR DESCRIPTION
**Pull request by mceachen**
* Upgraded ts-related deps with `salita --update`
* Explicitly added the new strict parameters to tsconfig
* The only thing that seemed to be broken was Mixin. Updated the index.d.ts and example.ts
* Ran `prettier` over the index.d.ts (it was missing a bunch of semis and indent was incorrect). This is isolated in a commit.

@kblomster if you have time, can you take a look?

### CODE REVIEW

File: typings/objection/index.d.ts [hunk 1]
```diff
@@ -106,7 +108,7 @@ declare namespace Objection {
 
   export interface RelationMapping {
     relation: Relation;
-    modelClass: typeof Model | string;
+    modelClass: ModelClass<any> | string;
```
Comment (mceachen) [inline typings/objection/index.d.ts]: this and `Mixin` were the only two consequential updates here. Everything else is from `prettier`.

**Comment by koskimas** [comment]
LGTM. Merge away 👍

File: typings/objection/index.d.ts [hunk 2]
```diff
@@ -64,8 +64,8 @@ declare namespace Objection {
     // than an identity function type. <M extends typeof Model> retains the
     // model subclass type in the return value, without requiring the user
     // to type the Mixin call.
```
Comment (kblomster) [inline typings/objection/index.d.ts]: This comment is obsolete and no longer matches the code below. Consider removing?

**Comment by kblomster** [comment]
You already merged but I didn't really have anything to add other than the comment anyway. :)

File: typings/objection/index.d.ts (hunk 2 above)
Comment (mceachen) [reply typings/objection/index.d.ts]: will do

**Pull request merged**